### PR TITLE
reference/sql: add show table regions docs

### PR DESCRIPTION
**Pull request by crazycs520**
### What is changed, added or deleted? 

* add docs for show table regions docs.

Here I only add doc in master, after I got LGTM, I will change the related v3.0 and v2.1 document.

### What is the related PR or file link(s)? 
https://github.com/pingcap/tidb/pull/10612

### CODE REVIEW

File: dev/reference/sql/statements/show-table-regions.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+5 rows in set
+-- 用 SPLIT TABLE REGION 语法切分索引数据的 REGION，下面语法把表 t 的索引 name 数据在 [a,z] 范围内切分成 2 个 REGION。
+test> split table t index name between ("a") and ("z") regions 2;
++--------------------+----------------------+
+| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
++--------------------+----------------------+
+| 2                  | 1.0                  |
++--------------------+----------------------+
+1 row in set
+-- 现在表 t  一共有 6 个 REGIONS，其中 ID 是 94 的 REGION 既会写入 [-inf,20000) 的行数据，也会写入索引 [l,+inf] 的 name 索引数据。
```
Comment (crazycs520) [inline dev/reference/sql/statements/show-table-regions.md]: 这里应该 要有 7 个  region 合适些，后面需要修复下 TiDB 这个行为。

Comment (winkyao) [reply dev/reference/sql/statements/show-table-regions.md]: 能否先修复再合 doc？

Comment (crazycs520) [reply dev/reference/sql/statements/show-table-regions.md]: ok, See PR: https://github.com/pingcap/tidb/pull/11721

File: dev/reference/sql/statements/show-table-regions.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+---
+title: SHOW TABLE REGIONS
+summary: TiDB 数据库中 SHOW TABLE REGIONS 的使用概况。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
summary: 了解如何使用 TiDB 数据库中的 SHOW TABLE REGIONS。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+category: reference
+---
+
+# SHOW TABLE REGIONS
+
+`SHOW TABLE REGIONS` 语句用于显示 TiDB 中某个表的 REGION 信息。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
`SHOW TABLE REGIONS` 语句用于显示 TiDB 中某个表的 REGION 信息。
```
```suggestion
`SHOW TABLE REGIONS` 语句用于显示 TiDB 中某个表的 Region 信息。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+## 语法图
+
+```sql
+index_name
+SHOW TABLE [table_name] REGIONS;
+
+SHOW TABLE [table_name] INDEX [index_name] REGIONS;
+```
+
+`SHOW TABLE REGIONS` 回返回如下列：
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
`SHOW TABLE REGIONS` 会返回如下列：

```

File: dev/reference/sql/statements/show-table-regions.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+* REGION_ID: REGION 的 ID。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `REGION_ID`：Region 的 ID。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+* START_KEY: REGION 的 START KEY。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `START_KEY`：Region 的 Start key。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+* END_KEY: REGION 的 END KEY。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `END_KEY`：Region 的 End key。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+* LEADER_ID: REGION 的 LEADER ID。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `LEADER_ID`：Region 的 Leader ID。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+* LEADER_STORE_ID: REGION LEADER 所在的 store (TiKV) ID。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `LEADER_STORE_ID`：Region leader 所在的 store (TiKV) ID。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+* PEERS: REGION 所有副本的 ID。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `PEERS`：Region 所有副本的 ID。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+* SCATTERING: REGION 是否正在打散或调度中。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* `SCATTERING`：Region 是否正在打散或调度中。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+## 示例
+
+```sql
+test> create table t (id int key,name varchar(50), index (name));
+Query OK, 0 rows affected
+-- 默认新键表后会单独 SPLIT 出一个 REGION 来存放这个这个表的数据，开始时行数据和索引数据都会写到这个 REGION 。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
-- 默认新建表后会单独 split 出一个 Region 来存放该表的数据，开始时行数据和索引数据都会写到这个 Region。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+test> show table t regions;
++-----------+-----------+---------+-----------+-----------------+-----------+------------+
+| REGION_ID | START_KEY | END_Key | LEADER_ID | LEADER_STORE_ID | PEERS     | SCATTERING |
++-----------+-----------+---------+-----------+-----------------+-----------+------------+
+| 5         | t_43_     |         | 8         | 2               | 8, 14, 93 | 0          |
++-----------+-----------+---------+-----------+-----------------+-----------+------------+
+1 row in set
+-- 用 SPLIT TABLE REGION 语法切分行数据的 REGION，下面语法把表 t 的行数据切分成 5 个 REGION。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
-- 用 `SPLIT TABLE REGION` 语法切分行数据的 Region，下面语法把表 t 的行数据切分成 5 个 Region。 
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+| REGION_ID | START_KEY    | END_Key      | LEADER_ID | LEADER_STORE_ID | PEERS         | SCATTERING |
++-----------+--------------+--------------+-----------+-----------------+---------------+------------+
+| 94        | t_43_        | t_43_r_20000 | 95        | 2               | 95, 96, 97    | 0          |
+| 98        | t_43_r_20000 | t_43_r_40000 | 101       | 4               | 101, 103, 102 | 0          |
+| 104       | t_43_r_40000 | t_43_r_60000 | 108       | 3               | 105, 106, 108 | 0          |
+| 109       | t_43_r_60000 | t_43_r_80000 | 111       | 13              | 111, 112, 113 | 0          |
+| 5         | t_43_r_80000 |              | 8         | 2               | 8, 14, 93     | 0          |
++-----------+--------------+--------------+-----------+-----------------+---------------+------------+
+5 rows in set
+-- 用 SPLIT TABLE REGION 语法切分索引数据的 REGION，下面语法把表 t 的索引 name 数据在 [a,z] 范围内切分成 2 个 REGION。
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
-- 用 `SPLIT TABLE REGION` 语法切分索引数据的 Region，下面语法把表 t 的索引 name 数据在 [a,z] 范围内切分成 2 个 Region。
```

File: dev/reference/sql/statements/show-table-regions.md (hunk 1 above)
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
-- 现在表 t 一共有 6 个 Region，其中 ID 是 94 的 Region 既会写入 [-inf,20000) 的行数据，也会写入索引 [l,+inf] 的 name 索引数据。
```

File: dev/reference/sql/statements/show-table-regions.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+| 109       | t_43_r_60000                | t_43_r_80000                | 111       | 13              | 111, 112, 113 | 0          |
+| 5         | t_43_r_80000                |                             | 8         | 2               | 8, 14, 93     | 0          |
+| 122       | t_43_i_1_                   | t_43_i_1_016d80000000000000 | 125       | 4               | 123, 124, 125 | 0          |
++-----------+-----------------------------+-----------------------------+-----------+-----------------+---------------+------------+
+6 rows in set
+```
+
+## 另请参阅
+
+* [SPLIT TABLE REGIONS](/reference/sql/statements/split-regions.md)
```
Comment (lilin90) [inline dev/reference/sql/statements/show-table-regions.md]:
```suggestion
* [SPLIT REGIONS](/reference/sql/statements/split-regions.md)
```

File: dev/TOC.md [hunk 16]
```diff
@@ -201,6 +201,7 @@
       - [`SHOW TABLE STATUS`](reference/sql/statements/show-table-status.md)
       - [`SHOW [GLOBAL|SESSION] VARIABLES`](reference/sql/statements/show-variables.md)
       - [`SHOW WARNINGS`](reference/sql/statements/show-warnings.md)
+      - [`SHOW TABLE REGION`](reference/sql/statements/show-table-regions.md)
```
Comment (lilin90) [inline dev/TOC.md]:
```suggestion
      - [`SHOW TABLE REGIONS`](reference/sql/statements/show-table-regions.md)
```

**Comment by lilin90** [comment]
@crazycs520 Please update the related `v3.0` and `v2.1` documents. Thanks!

**Comment by lilin90** [comment]
@TomShawn When you align docs with changes in this PR, please also combine #1788. Note that `split-region.md` is added to `v2.1`. Thanks!

**Comment by TomShawn** [comment]
> @TomShawn When you align docs with changes in this PR, please also combine #1788. Note that `split-region.md` is added to `v2.1`. Thanks!

Got it!

**Pull request merged**